instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb

```diff
@@ -50,6 +50,7 @@ def query(sql)
               ),
               kind: :client
             ) do |_span, context|
+              sql = sql.dup unless !sql.frozen? || propagator.is_a?(OpenTelemetry::Instrumentation::Trilogy::NoopPropagator)
```

Comment:
I think we could do 

```ruby
sql = +sql unless propagator.is_a?(OpenTelemetry::Instrumentation::Trilogy::NoopPropagator)
```

But I think the syntax is slightly obscure ([docs](https://ruby-doc.org/3.2.2/String.html#method-i-2B-40)). Curious for others' thoughts.

```ruby
irb(main):001:0> x = "foo".freeze
=> "foo"
irb(main):002:0> x.gsub!("f", "o")
Traceback (most recent call last):
        5: from /usr/bin/irb:23:in `<main>'
        4: from /usr/bin/irb:23:in `load'
        3: from /Library/Ruby/Gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        2: from (irb):2
        1: from (irb):2:in `gsub!'
FrozenError (can't modify frozen String)
irb(main):003:0> x = +x
=> "foo"
irb(main):004:0> x.gsub!("f", "o")
=> "ooo"
irb(main):005:0> x
=> "ooo"
```

Reply:
Are we mutating the string to avoid additional allocations?

Reply:
Yeah, sorry, got a little too steeped in my own context... Updated pr description with:

> Why do this? We may want to propagate context into sql strings that arrive at the trilogy client frozen.

Does that clarify things?

Reply:
I think it may be cleaner at this point to not install a noop propagator and instead check `unless propagator.nil?` here.

Reply:
I agree; will make this change